### PR TITLE
add xvega

### PR DESCRIPTION
**Pull request by madhur-tandon**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/xvega/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  url: https://github.com/QuantStack/xvega/archive/{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [win and vc<14]
+
+requirements:
+  build:
+    - cmake
```
Comment (scopatz) [inline recipes/xvega/meta.yaml:20]: please add make as a dep on unix

File: recipes/xvega/build.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+cmake -DBUILD_TESTS=OFF -DCMAKE_INSTALL_PREFIX=$PREFIX $SRC_DIR -DCMAKE_INSTALL_LIBDIR=lib
+make install
```
Comment (scopatz) [inline recipes/xvega/build.sh:4]: please add `- j ${CPU_COUNT}`

File: recipes/xvega/build.sh [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+cmake -DBUILD_TESTS=OFF -DCMAKE_INSTALL_PREFIX=$PREFIX $SRC_DIR -DCMAKE_INSTALL_LIBDIR=lib
```
Comment (scopatz) [inline recipes/xvega/build.sh:3]: Also, this should be a release build

File: recipes/xvega/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    - test -f ${PREFIX}/include/xvega/utils/serialize.hpp  # [unix]
+    - test -f ${PREFIX}/lib/cmake/xvega/xvegaConfig.cmake  # [unix]
+    - test -f ${PREFIX}/lib/cmake/xvega/xvegaConfigVersion.cmake  # [unix]
+    - if exist %LIBRARY_PREFIX%\include\xvega\xvega.hpp (exit 0) else (exit 1)  # [win]
+    - if exist %LIBRARY_PREFIX%\include\xvega\base\xvega-base.hpp (exit 0) else (exit 1)  # [win]
+    - if exist %LIBRARY_PREFIX%\include\xvega\functions\populate_data.hpp (exit 0) else (exit 1)  # [win]
+    - if exist %LIBRARY_PREFIX%\include\xvega\grammar\marks.hpp (exit 0) else (exit 1)  # [win]
+    - if exist %LIBRARY_PREFIX%\include\xvega\utils\serialize.hpp (exit 0) else (exit 1)  # [win]
+    - if exist %LIBRARY_PREFIX%\lib\cmake\xvegaConfig.cmake (exit 0) else (exit 1)  # [win]
+    - if exist %LIBRARY_PREFIX%\lib\cmake\xvegaConfigVersion.cmake (exit 0) else (exit 1)  # [win]
```
Comment (scopatz) [inline recipes/xvega/meta.yaml:50]: Also, please ensure that you are not building static libraries

File: recipes/xvega/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE
+  summary: A C++ backend for vega-lite
+  description: vega-lite visualization system for the xeus-cling kernel
+  dev_url: https://github.com/QuantStack/xvega
+
+extra:
+  recipe-maintainers:
+    - SylvainCorlay
```
Comment (scopatz) [inline recipes/xvega/meta.yaml:63]: @SylvainCorlay are you willing to be a maintainer?

File: recipes/xvega/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/QuantStack/xvega/archive/{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: true  # [win and vc<14]
```
Comment (scopatz) [inline recipes/xvega/meta.yaml:16]: This should have a run_exports

File: recipes/xvega/meta.yaml (hunk 5 above)
Comment (SylvainCorlay) [reply recipes/xvega/meta.yaml:63]: +1

File: recipes/xvega/meta.yaml (hunk 6 above)
Comment (SylvainCorlay) [reply recipes/xvega/meta.yaml:16]: xvega is header-only, so probably not at this point.

File: recipes/xvega/meta.yaml (hunk 4 above)
Comment (SylvainCorlay) [reply recipes/xvega/meta.yaml:50]: (header only)

File: recipes/xvega/meta.yaml (hunk 1 above)
Comment (SylvainCorlay) [reply recipes/xvega/meta.yaml:20]: +1

File: recipes/xvega/meta.yaml (hunk 6 above)
Comment (scopatz) [reply recipes/xvega/meta.yaml:16]: Oh ok

File: recipes/xvega/build.sh (hunk 3 above)
Comment (scopatz) [reply recipes/xvega/build.sh:3]: Even if it is header-only it is a good idea to make this a release build in case of future changes

Comment (SylvainCorlay) [reply recipes/xvega/build.sh:3]: good point.

**Comment by SylvainCorlay** [comment]
Thanks for the thorough review @scopatz.

**Pull request merged**